spring-integration-file/src/main/java/org/springframework/integration/file/FileWritingMessageHandler.java

```diff
@@ -305,8 +302,11 @@ protected Object handleRequestMessage(Message<?> requestMessage) {
 						(StringUtils.hasText(this.temporaryFileSuffix) && tempFile.exists()));
 
 		if (!ignore) {
-
 			try {
+				if (!resultFile.exists()) {
+					tempFile.getParentFile().mkdirs();
+					resultFile.getParentFile().mkdirs();
```

Comment:
Yeah, agreed, no reason:
```
File tempFile = new File(destinationDirectoryToUse, generatedFileName + this.temporaryFileSuffix);
File resultFile = new File(destinationDirectoryToUse, generatedFileName);
```

Let me know if you see anything else here to fix and push.